src/common/server.ts

```diff
@@ -39,17 +39,34 @@ async function createExperimentalServer(
   outputChannel: LogOutputChannel,
   initializationOptions: IInitOptions,
 ): Promise<LanguageClient> {
-  const command = RUFF_BIN_PATH;
-  const cwd = settings.cwd;
-  const args = [RUFF_SERVER_CMD, ...RUFF_SERVER_REQUIRED_ARGS];
-
-  const serverOptions: ServerOptions = {
-    command,
-    args,
-    options: { cwd, env: process.env },
-  };
-
-  traceInfo(`Server run command: ${[command, ...args].join(" ")}`);
+  let serverOptions: ServerOptions;
+  // If the user provided a binary path, we'll try to call that path directly.
+  if (settings.path.length > 0 && settings.path[0]) {
```

Comment:
Nit: I think this would be slightly more idiomatic in JS because it's "safe" to access out of bound elements
```suggestion
  if (settings.path[0] != null) {
```

Reply:
Oh, cool! I was not aware of that.